compiler/src/iree/compiler/Codegen/Common/GPU/GPUVectorDistribution.cpp

```diff
@@ -132,7 +132,8 @@ void DistributionPattern::replaceOpWithDistributedValues(
   for (auto [opResult, replacement] :
        llvm::zip_equal(op->getOpResults(), values)) {
     // If this value is a vector type, it must be converted back to simd.
-    if (isa<VectorType>(replacement.getType())) {
+    if (isa<VectorType>(replacement.getType()) &&
+        cast<ShapedType>(replacement.getType()).getRank() != 0) {
```

Comment:
use dyn_cast instead:
```c++
if (auto x = dyn_cast<Y>(y)) {
  if (x.something() == Z) {
 ```
 
 See https://llvm.org/docs/ProgrammersManual.html#the-isa-cast-and-dyn-cast-templates:~:text=Note%20that%20you%20should%20not%20use%20an%20isa%3C%3E%20test%20followed%20by%20a%20cast%3C%3E%2C%20for%20that%20use%20the%20dyn_cast%3C%3E%20operator.